src/ResponseBodyHandler.cpp

```diff
@@ -46,13 +50,28 @@ void ResponseBodyHandler::execute()
 		}
 		m_request.httpStatus = StatusOK;
 		m_request.targetResource += "autoindex.html";
-	} else if (m_request.method == MethodGet) {
+		return;
+	} 
+	if (m_request.method == MethodGet) {
 		try {
 			m_responseBody = m_fileSystemPolicy.getFileContents(m_request.targetResource.c_str());
 		} catch (std::exception& e) {
 			m_request.httpStatus = StatusInternalServerError;
 			handleErrorBody();
 		}
+		return;
+	}
+	if (m_request.method == MethodPost) {
+		FileWriteHandler fileWriteHandler(m_fileSystemPolicy);
+		m_responseBody = fileWriteHandler.execute(m_request.targetResource, m_request.body);
+		if (m_responseBody.find("created") != std::string::npos) {
```

Comment:
I think this should check if the file was created or appended? Shouldn't a successful POST request always have Status 201? For newly created and appended files. I know an appended file is not really created, but 201 is the code we expect if POST was successful.

Reply:
According to [this](https://dev.to/msnmongare/handling-http-status-codes-in-a-rest-api-200-ok-vs-201-created-28d) and some other sources, a 201 code is solely used when a new resource is (successfully) created, whereas 200 indicates a successful action but which does not create a new resource (such as appending data to a file). Therefore I would only have the 201 code a new file was created by the POST request.

Reply:
Ok!